src/components/form/form.js

```diff
@@ -509,6 +523,10 @@ class Form extends React.Component {
    * @return {void}
    */
   handleOnSubmit = (ev) => {
+    if (this.props.autoDisable) {
+      this.setState({ submitted: true });
+    }
```

Comment:
if client side validation fails we could re-enable automatically, after

```
const valid = this.validate();
```

inside the `if (!valid)` check that is already there probably

Reply:
https://github.com/Sage/carbon/blob/0268cd392a50784427e553493af965064b58a8c8/src/components/form/form.js#L549 should it go here ?

Reply:
I thought you had put it here? https://github.com/Sage/carbon/pull/1688/files#diff-ce2b1986d94d1af0464dc730fd2bbe29R539

(my original comment was from 7 days ago)